#! /bin/sh /usr/share/dpatch/dpatch-run
## 55_pgrep_usage_exitcode.dpatch by  <hesso@pool.math.tu-berlin.de>
##
## DP: Distinguish between invalid commandline parameters and '-?'.

@DPATCH@
--- procps-3.2.7.orig/pgrep.c	2007-08-06 16:35:30.000000000 +0200
+++ procps-3.2.7/pgrep.c	2007-08-06 16:35:06.000000000 +0200
@@ -678,7 +678,7 @@
 //		case 'z':   // Solaris: match by zone ID
 //			break;
 		case '?':
-			usage (opt);
+			usage (optopt?optopt:opt);
 			break;
 		}
 	}
